### PR TITLE
User/derek meek/offline cluster supportb

### PR DESCRIPTION
**Pull request by DerekMeek**

### CODE REVIEW

File: PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1 [hunk 1]
```diff
@@ -2674,8 +2718,8 @@ function Get-SummaryReport
 
     $ClusterNodes = Import-Clixml (Join-Path $Path "GetClusterNode.XML")
     $Cluster = Import-Clixml (Join-Path $Path "GetCluster.XML")
-
-    $ClusterName = $Cluster.Name + "." + $Cluster.Domain
+	
+    $ClusterName = $Cluster.Name + "." + $ClusterDomain
```
Comment (dl2n) [inline PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1]: Where is ClusterDomain getting set during the SummaryReport? Try running Show-SddcDiagnosticReport on the produced ZIP package, I suspect this is relying on dynamic scoping when run during Get's invocation to create the summary report.

File: PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1 [hunk 2]
```diff
@@ -839,24 +839,45 @@ function Get-SddcDiagnosticInfo
...
+		$Cluster | Export-Clixml ($Path + "GetCluster.XML")
+		$ClusterName = $Cluster.Name + "." + $Cluster.Domain
+		$S2DEnabled = $Cluster.S2DEnabled
+		$ClusterDomain = $Cluster.Domain 
+	}
+	else
+	{
+		# We can only get here if -Nodelist was used, but cluster service isn't running
+		Write-Error "Cluster service was not running on any node, some information will be unavailable"
+		$ClusterName = "UNAVAILABLE";
```
Comment (dl2n) [inline PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1]: I'd be strongly inclined to use empty/null string v. a special value. Render as "UNAVAILABLE" when we're emitting it into reports, as needed. May need a more verbose comment though, such as indicating it is unavailable due to the cluster being down, etc. This is where we should export a recoverable indication of the state of the cluster so the summary report can repeat it. In the cluster-up case, we get GetCluster.XML.

File: PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1 [hunk 3]
```diff
@@ -839,25 +839,47 @@ function Get-SddcDiagnosticInfo
...
-    $S2DEnabled = $Cluster.S2DEnabled
-
-    # Select an access node, which will be used to query the cluster
-
-    $AccessNode = ($ClusterNodes)[0].Name + "." + $Cluster.Domain
-    Write-Host "Cluster name               : $ClusterName"
+	
+	if ($Cluster -ne $null)
+	{
+		$Cluster | Export-Clixml ($Path + "GetCluster.XML")
```
Comment (dl2n) [inline PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1]: Note GetCluster.XML is only dropped if we find a cluster.

Comment (DerekMeek) [reply PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1]: Yup, doesn't make sense to drop a file we couldn't possible create.

File: PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1 [hunk 4]
```diff
@@ -2674,7 +2719,7 @@ function Get-SummaryReport
 
     $ClusterNodes = Import-Clixml (Join-Path $Path "GetClusterNode.XML")
     $Cluster = Import-Clixml (Join-Path $Path "GetCluster.XML")
```
[the remark this answers is not in the document]
Comment (DerekMeek) [reply PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1]: With these files we cannot drop when cluster is unavailable does that result in $null for the result of Import-Clixml? what do you want to do in this function when these values are null?

[the remark this answers is not in the document]
Comment (dl2n) [reply PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1]: See what happens if we -ErrorAction SilentlyContinue. I suspect that'll give us a clean $null we can then use safely the rest of the way w/o having to do an explicit Test-Path or other direct check for file existence.

**Comment by DerekMeek** [comment]
Yup, that worked.

**Pull request merged**